.github/workflows/labels.yml

```diff
@@ -64,6 +79,21 @@ jobs:
               artifacts: 0
             }
 
+            // We have 12,500 requests with the app token. Other jobs only need a minimum of probably less than 100.
+            // We limit scheduled runs to 500 requests / hour. The schedule currently runs 6x / hour.
+            // Even though this would theoretically allow 6*500 = 3,000 requests, in practice this should not happen:
+            // Due to the concurrency group, there will be a single scheduled run maximum. Thus, if one of them hits
+            // the limit, the others will not use more requests.
+            // For manual runs, we go up as high as 10k, which should allow multiple days of backfilling over a few hours.
+            const reservoir = context.eventName == 'workflow_dispatch' ? 10000 : 500
+
+            // As a safe-guard, we check the remaining requests. We'll always want to keep a very conservative number of
+            // 1000 requests free for regular operation of other jobs. Running in the context of a pull request is
+            // always allowed, we don't expect many requests anyway.
+            const { remaining } = (await github.rest.rateLimit.get()).data.resources.core
+            if (!context.payload.pull_request && remaining < reservoir + 1000)
+              throw new Error(`Not enough API requests remaining to proceed: ${remaining}.`)
```

Comment:
Could we do this in a while loop to potentially recover from being close to our self-imposed limit?

```js
// Psudocode
while (remaining < reservoir + 1000) {
    remaining -= 100
    if (remaining < 5) {
        throw new Error("....")
    }
}
log(`using reservoir ${reservoir} with remaining ${remaining}`)
```

Reply:
Or avoid the loop with something like:

```js
if (remaining < reservoir + 1000) {
     reservoir = remaining - 1000
}

if (reservoir < 5) {
    throw new Error("...")
}
```